proto/cluster.proto

```diff
@@ -145,34 +146,36 @@ message Cluster {
   Environment environment = 5;
   
   // Required field. This field indicates ray cluster configuration
-  ClusterSpec cluster_spec = 6;
+  ClusterSpec cluster_spec = 6 [(google.api.field_behavior) = REQUIRED];
+ 
   // Optional. Annotations, for example, "kubernetes.io/ingress.class" to define Ingress class 
-  map<string, string> annotations = 7;
+  map<string, string> annotations = 7 [(google.api.field_behavior) = OPTIONAL];
 
   // Optional input field. Container environment variables from user.
-  map<string, string> envs = 8;
+  map<string, string> envs = 8 [(google.api.field_behavior) = OPTIONAL];
 
   // Output. The time that the cluster created.
-  google.protobuf.Timestamp created_at = 9;
+  google.protobuf.Timestamp created_at = 9 [(google.api.field_behavior) = OUTPUT_ONLY];
 
   // Output. The time that the cluster deleted.
-  google.protobuf.Timestamp deleted_at = 10;
+  google.protobuf.Timestamp deleted_at = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
 
   // Output. The status to show the cluster status.state
-  string cluster_state = 11;
+  string cluster_state = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
 
   // Output. The list related to the cluster.
-  repeated ClusterEvent events = 12;
+  repeated ClusterEvent events = 12 [(google.api.field_behavior) = OUTPUT_ONLY];
   
   // Output. The service endpoint of the cluster
-  map<string, string> service_endpoint = 13;
+  map<string, string> service_endpoint = 13 [(google.api.field_behavior) = OUTPUT_ONLY];
 }
 
+// Cluster specification.
 message ClusterSpec {
-  // The head group configuration
-  HeadGroupSpec head_group_spec = 1;
-  // The worker group configurations
-  repeated WorkerGroupSpec worker_group_spec = 2;
+  // Required. The head group configuration
+  HeadGroupSpec  head_group_spec = 1[(google.api.field_behavior) = REQUIRED];
+  // Required. The worker group configurations
+  repeated WorkerGroupSpec worker_group_spec = 2 [(google.api.field_behavior) = REQUIRED];
```

Comment:
I think `WorkerGroupSpec` could be optional.
https://github.com/z103cb/kuberay/blob/23303f4902cc2a37da28653ff637b32fe81c511f/helm-chart/kuberay-operator/crds/ray.io_rayclusters.yaml#L6651

Reply:
@tedhtchang, for I think it should be required. There are these validations in place:


[ValidateCreateClusterRequest](https://github.com/z103cb/kuberay/blob/23303f4902cc2a37da28653ff637b32fe81c511f/apiserver/pkg/server/cluster_server.go#L139)

```golang
	for index, spec := range request.Cluster.ClusterSpec.WorkerGroupSpec {
		if len(spec.GroupName) == 0 {
			return util.NewInvalidInputError("WorkerNodeSpec %d group name is empty. Please specify a valid value.", index)
		}
		if len(spec.ComputeTemplate) == 0 {
			return util.NewInvalidInputError("WorkerNodeSpec %d compute template is empty. Please specify a valid value.", index)
		}
		if spec.MaxReplicas == 0 {
			return util.NewInvalidInputError("WorkerNodeSpec %d MaxReplicas can not be 0. Please specify a valid value.", index)
		}
		if spec.MinReplicas > spec.MaxReplicas {
			return util.NewInvalidInputError("WorkerNodeSpec %d MinReplica > MaxReplicas. Please specify a valid value.", index)
		}
	}
```

Reply:
@tedhtchang I have validated that this this field is indeed optional. Can you review this again ?

Reply:
It is optional

Reply:
Fixed in [be7ffb6](https://github.com/ray-project/kuberay/commit/be7ffb60e882aca9d61f887e98b49914f2ce61ce)